[ONNX] Add sanity check in CI for onnxbench

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114907
* __->__ #110178

ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.

BowenBao committed:
[ONNX] Add sanity check in CI for onnxbench

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

thiagocrepaldi commented on scripts/onnx/test.sh:
```diff
@@ -48,6 +48,41 @@ if [[ "$SHARD_NUMBER" == "2" ]]; then
+  device="cpu"
+
+  mkdir -p "${log_folder}"
+  for mode in "accuracy" "performance"; do
+    for compiler in "dynamo-onnx" "torchscript-onnx"; do
+      for suite in "huggingface" "timm_models"; do
```
use variables to hold `mode`, `compiler` and `suite` list values? makes it easier to extend later

thiagocrepaldi reviewed (commented):
not blocking for me but @kit1980 and @malfet once advised us to move CI script logic to Python instead of shell to make it cross platform and less messy due to compatibility issues (bash vs zshell vs etc)

BowenBao committed:
rebase on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
rebase on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

BowenBao committed:
address comment on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]

thiagocrepaldi commented on scripts/onnx/test.sh:
```diff
@@ -48,6 +48,44 @@ if [[ "$SHARD_NUMBER" == "2" ]]; then
+  # Sanity check on torchbench w/ onnx
+  pip install pandas
+  log_folder="test/.torchbench_logs"
+  device="cpu"
```
Do we still want `cpu` only as device or should we add a `for device in "$devices[@]}"; do`?

BowenBao replied:
cpu only since CI for us is cpu only

thiagocrepaldi commented on scripts/onnx/test.sh:
```diff
@@ -48,6 +48,44 @@ if [[ "$SHARD_NUMBER" == "2" ]]; then
+            "timm_models")
+                bench_args+=(-k lcnet_050)
+                ;;
+            *)
```
just confirming we don't have anything for "ado")

BowenBao replied:
no we don't, this is running in our github onnx ci.

thiagocrepaldi reviewed (approved):
LGTM assuming we only care about cpu and no ADO suite

BowenBao committed:
Update on "[ONNX] Add sanity check in CI for onnxbench"


ONNX CI to run benchmark with `--quick` to validate the onnxbench infra.


[ghstack-poisoned]
